### PR TITLE
Allow mailingLists in traits

### PR DESCRIPTION
**Pull request by danrowden**
This PR modifies the mailingLists field from an array of objects to a single object. Since users can't manually provide objects for the array, the previous structure was not very useful. It's unlikely that event payloads would include an array of mailingLists objects by default.

With this update, users can now easily input multiple mailling list IDs when creating a mapping

`mailingLists` added to traits can also be sent to the APIm with our without manual mapping in the UI.

### CODE REVIEW

File: packages/destination-actions/src/destinations/loops/createOrUpdateContact/index.ts [hunk 1]
```diff
@@ -146,10 +122,27 @@ const action: ActionDefinition<Settings, Payload> = {
...
+    if (typeof obj[key] !== 'boolean') {
+      if (obj[key] === 'true') {
+        result[key] = true
+      } else if (obj[key] === 'false') {
+        result[key] = false
+      }
+    } else {
+      result[key] = obj[key]
+    }
+  }
```
Comment (tcgilbert) [inline packages/destination-actions/src/destinations/loops/createOrUpdateContact/index.ts:146]:
```suggestion
for (const key in obj) {
    const value = obj[key];
    
    if (typeof value === 'string') {
      if (value === 'true') {
        result[key] = true;
      } else if (value === 'false') {
        result[key] = false;
      }
    } else if (typeof value === 'boolean') {
      result[key] = value;
    }
  }
```

Getting a type issue here with the build, perhaps try the above^: 
Screenshot 2024-09-27 at 1 20 33 PM

**Review by tcgilbert — CHANGES_REQUESTED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @tcgilbert will you be reviewing this?

**Review by tcgilbert — APPROVED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @danrowden and @tcgilbert - PR deployed. Please confirm you are happy with the update.

**Comment by danrowden** [comment]
@joe-ayoub-segment Yes, happy thanks! I've tested it and it works

**Pull request merged**